src/chart/generateCategoricalChart.tsx

```diff
@@ -2304,22 +2338,25 @@ export const generateCategoricalChart = ({
 
       const events = this.parseEventsOfWrapper();
       return (
-        <div
-          className={clsx('recharts-wrapper', className)}
-          style={{ position: 'relative', cursor: 'default', width, height, ...style }}
-          {...events}
-          ref={(node: HTMLDivElement) => {
-            this.container = node;
-          }}
-          role="region"
-        >
-          <Surface {...attrs} width={width} height={height} title={title} desc={desc} style={FULL_WIDTH_AND_HEIGHT}>
-            {this.renderClipPath()}
-            {renderByOrder(children, this.renderMap)}
-          </Surface>
-          {this.renderLegend()}
-          {this.renderTooltip()}
-        </div>
+        <ChartLayoutContextContainer>
+          <div
+            className={clsx('recharts-wrapper', className)}
+            style={{ position: 'relative', cursor: 'default', width, height, ...style }}
+            {...events}
+            ref={(node: HTMLDivElement) => {
+              this.container = node;
+            }}
+            role="region"
+          >
+            <Surface {...attrs} width={width} height={height} title={title} desc={desc} style={FULL_WIDTH_AND_HEIGHT}>
+              {this.renderClipPath()}
+              {renderByOrder(children, this.renderMap)}
+            </Surface>
+            {this.renderLegend()}
+            {this.renderTooltip()}
+            {this.updateLayoutContext()}
```

Comment:
I'm concerned about updating the context within the context. 
As it stands, It feels like adding unnecessary rendering behaviors.

Reply:
That's right. What can I do instead to update the context without the double render?

Reply:
Is it acceptable to directly pass value to the Provider without using useState?

Reply:
I'll try that

Reply:
Done, what do you think @HHongSeungWoo ?

Reply:
I think it's good!